reference-data-proxy/src/v1/controllers/acbs.controller.ts

```diff
@@ -79,78 +80,98 @@ export const findOne = async (req: Request, res: Response) => {
+          deal,
+          bank,
+        },
+      });
+    } catch (error: any) {
+      console.error('Error creating ACBS record: ', { error });
+      return null;
+    }
   }
-  return {};
 };
 
 export const createAcbsRecordPOST = async (req: Request, res: Response) => {
   if (req) {
     try {
       const { deal, bank } = req.body;
-      const { status, data } = await createAcbsRecord(deal, bank);
-      return res.status(status).send(data);
-    } catch (error) {
-      console.error('ACBS post failed ', { error });
+      const response = await createAcbsRecord(deal, bank);
+      if (response) {
+        const { status, data } = response;
+        return res.status(status).send(data);
+      }
+    } catch (error: any) {
+      console.error('ACBS create POST failed ', { error });
+      return res.status(400).send();
     }
   }
   return res.status(400).send();
 };
 
 const issueAcbsFacility = async (id: any, facility: object, deal: object) => {
   if (id) {
-    const response = await axios({
-      method: 'post',
-      url: `${acbsFunctionUrl}/api/orchestrators/acbs-issue-facility`,
-      data: {
-        facilityId: id,
-        facility,
-        deal,
-      },
-    }).catch((err: any) => err);
-    return response;
+    try {
+      await axios({
+        method: 'post',
+        url: `${acbsFunctionUrl}/api/orchestrators/acbs-issue-facility`,
+        data: {
+          facilityId: id,
+          facility,
+          deal,
+        },
+      });
+    } catch (error: any) {
+      console.error('ACBS issue facility POST failed ', { error });
+      return null;
+    }
   }
-  return {};
 };
 
 export const issueAcbsFacilityPOST = async (req: Request, res: Response) => {
-  if (req) {
-    const { id } = req.params;
-    const { facility, deal } = req.body;
-    const { status, data } = await issueAcbsFacility(id, facility, deal);
-    return res.status(status).send(data);
+  try {
+    if (req) {
+      const { id } = req.params;
+      const { facility, deal } = req.body;
+      const response = await issueAcbsFacility(id, facility, deal);
+      if (response) {
+        const { status, data } = response;
+        return res.status(status).send(data);
+      }
+    }
+    return res.status(400).send();
+  } catch (e) {
+    console.error('Error during ACBS facility issue POST: ', { e });
+    return res.status(400).send();
   }
-  return res.status(400).send();
 };
 
 /**
  * Invoked Azure DOF using HTTP `POST` method.
  * @param {Object} amendment Amendment object comprising facility ID and amends. A amendment at a time is processed.
  * @returns {Object} DOF Response
  */
-const amendAcbsFacility = async (amendment: object) => {
-  const hasAcceptablePayload = Object.prototype.hasOwnProperty.call(amendment, 'coverEndDate') || Object.prototype.hasOwnProperty.call(amendment, 'amount');
+const amendAcbsFacility = async (amendment: Amendment) => {
+  const hasAcceptablePayload = amendment.coverEndDate || amendment.amount;
```

Comment:
if we leave it here, can we rename it into something like: `hasAmendment`?

Reply:
Updated.